cmd/influx/write_test.go

```diff
@@ -0,0 +1,436 @@
+			if err != nil {
+				log.Fatal("Unable to create gzip reader", err)
+				return
+			}
+			requestData = gzipReader
+		}
+		lineData, _ = ioutil.ReadAll(requestData)
+		rw.Write([]byte(`OK`))
+	}))
+	defer server.Close()
+
+	// validation: --org or --org-id must be specified
+	lineData = lineData[:0]
+	command := cmdWrite(&globalFlags{}, genericCLIOpts{w: ioutil.Discard})
+	command.SetArgs([]string{"--format", "csv"})
+	err := command.Execute()
+	require.Contains(t, fmt.Sprintf("%s", err), "org")
+
+	// validation: either --bucket or --bucket-id must be specified
+	lineData = lineData[:0]
+	command = cmdWrite(&globalFlags{}, genericCLIOpts{w: ioutil.Discard})
+	command.SetArgs([]string{"--format", "csv", "--org", "my-org", "--bucket", "my-bucket", "--bucket-id", "my-bucket"})
+	err = command.Execute()
+	require.Contains(t, fmt.Sprintf("%s", err), "bucket") // bucket or bucket-id, but not both
+
+	// validation: unsupported --precision
+	lineData = lineData[:0]
+	command = cmdWrite(&globalFlags{}, genericCLIOpts{w: ioutil.Discard})
+	command.SetArgs([]string{"--format", "csv", "--org", "my-org", "--bucket", "my-bucket", "--precision", "pikosec"})
+	err = command.Execute()
+	require.Contains(t, fmt.Sprintf("%s", err), "precision") // invalid precision
+
+	// validation: --host must be supplied
+	prevHost := flags.Host
+	prevToken := flags.Token
+	flags.Host = ""
+	flags.Token = "myToken"
+	defer func() {
+		flags.Host = prevHost
+		flags.Token = prevToken
+	}()
+	lineData = lineData[:0]
+	command = cmdWrite(&globalFlags{}, genericCLIOpts{w: ioutil.Discard})
+	command.SetArgs([]string{"--format", "csv", "--org", "my-org", "--bucket", "my-bucket"})
+	err = command.Execute()
+	require.Contains(t, fmt.Sprintf("%s", err), "host")
+
+	// validation: failed to decode bucket-id
+	lineData = lineData[:0]
+	flags.Host = server.URL
+	command = cmdWrite(&globalFlags{}, genericCLIOpts{w: ioutil.Discard})
+	command.SetArgs([]string{"--format", "csv", "--org", "my-org", "--bucket-id", "my-bucket"})
+	err = command.Execute()
+	require.Contains(t, fmt.Sprintf("%s", err), "bucket-id")
+
+	// validation: failed to decode org-id
+	lineData = lineData[:0]
+	flags.Host = server.URL
+	command = cmdWrite(&globalFlags{}, genericCLIOpts{w: ioutil.Discard})
+	command.SetArgs([]string{"--format", "csv", "--org-id", "my-org", "--bucket", "my-bucket"})
+	err = command.Execute()
+	require.Contains(t, fmt.Sprintf("%s", err), "org-id")
+
+	// validation: failed to retrive buckets
+	lineData = lineData[:0]
+	flags.Host = server.URL
+	command = cmdWrite(&globalFlags{}, genericCLIOpts{w: ioutil.Discard})
+	// note: my-error-bucket parameter causes the test server to fail
+	command.SetArgs([]string{"--format", "csv", "--org", "my-org", "--bucket", "my-error-bucket"})
+	err = command.Execute()
+	require.Contains(t, fmt.Sprintf("%s", err), "bucket") // failed to retrieve buckets
+
+	// validation: no such bucket found
+	lineData = lineData[:0]
```

Comment:
are these sub tests? its easier to erad, but feels like a lot of coupling between assertions.  Can this be split up into sub tests?

```go
t.Run("validates no bucket found", func(t *testing.T) {
       lineData := lineData[:0]
        command := cmdWrite(&globalFlags{}, genericCLIOpts{w: ioutil.Discard})
	// note: my-empty-org parameter causes the test server to return no buckets
	command.SetArgs([]string{"--format", "csv", "--org", "my-empty-org", "--bucket", "my-bucket"})
	err = :command.Execute()
	require.Contains(t, fmt.Sprintf("%s", err), "bucket") // no such bucket found
})
```

same goes for a lot of these assertions here. They look like sub tests to me but aren't being treated as such. Please correct me if that assumption is wrong 🤔

Reply:
Yes, it looks better when using sub-tests.